transformer_engine/common/fused_attn/fused_attn_f16_arbitrary_seqlen.cu

```diff
@@ -311,10 +313,14 @@ void fused_attn_arbitrary_seqlen_fwd_impl(
           offset_v, offset_o, dropout_seed, dropout_offset] =
         get_graph(sdpa_f16_fprop_cache, descriptor);
 
-    auto plan_workspace_size = mha_graph->get_workspace_size();
     // Exit to request upper level API to allocate memory if needed
-    size_t actual_seqlen_workspace_size = 2 * b * sizeof(int32_t);
-    size_t seqlen_offsets_workspace_size = 4 * (b + 1) * sizeof(int32_t);
+    // n.b. Care should be taken to align each of the added worksapce tensors to their type.
+    // We do this by adding padding at the end of each separate allocation.
+    auto plan_workspace_size = alignTo<16>(mha_graph->get_workspace_size());
+    const size_t actual_seqlen_workspace_size = alignTo<16>(2 * b * sizeof(int32_t));
```

Comment:
How come we don't pad each of the two actual_seqlen tensors, like we did with the 4 ragged_offset tensors below?

Reply:
Its a good point I'll fix it so that each tensor gets aligned individually to be consistent and the most "safe".